### PR TITLE
#427 Make a graphical representation of the programs and activities on the frontend side

### LINKED ISSUE #427: The webapp should show a graphical representation of the programs and activities
## Description

Once the work for issue #415 is complete, we should create a very basic view of the programs and activities returned from the API request as a stepping stone to having the calendar in place.

## Details and resources

This does not need to be fancy. It should just show the formatted details of the default program and its activities (with time in the browser's time zone, not UTC). This is a necessary prerequisite for showing the data in the calendar.

## Checklist

- [x] This issue is linked to the appropriate project.
- [x] This issue has appropriate labels.

### PR DESCRIPTION
**Pull request by veranika-karpava**
Co-authored by: @wendyyng 
Co-authored by: @phoebehala 
Co-authored by: @veranika-karpava  

## Proposed changes

This pull request resolve #427. Created the Programs and Activities components that display the program data (default program and its activities) fetched from the program API. The Activities component acts as a stepping stone for displaying the program activities data in the calendar. It may be modified or replaced by calendar related components in future commits.

## Checklist
- [x] Are the issues being addressed linked to this PR?
- [x] Do all commit messages start with the issue number?
- [x] Are all code changes sufficiently tested?
- [x] Are there screenshots for UI changes?

Screenshot_2022-11-21_at_11 39 25_PM

### CODE REVIEW

File: webapp/src/components/CalendarPage.tsx [hunk 1]
```diff
@@ -1,7 +1,51 @@
...
+    isLoading,
+  } = useApiData<ProgramWithActivities[]>({
+    deps: [],
+    path: '/programs',
+    sendCredentials: true,
+  });
+  if (error) {
+    return <p>There was an error loading the programs.</p>;
+  }
+
```
Comment (seidior) [inline webapp/src/components/CalendarPage.tsx:21]: We can tighten up the whitespace on lines 21, 25, and 29.

**Review by seidior — APPROVED** [review]
I checked out the code on my development machine, recreated the Docker containers, brought the containers up, and tested it out.

The functionality seems to match the description of the changes, with the expected output being shown on the `/calendar` page in the webapp. The observed UI changes match the screenshot, and the code appears to conform to the code style used in similar files.

All automated tests pass without error, and yarn delint did not modify any files or show any errors. I approve of this pull request.

File: webapp/src/components/Activity.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+interface ActivityProps {
```
Comment (camianderson) [inline webapp/src/components/Activity.tsx:3]: Should we add this to the api.d.ts file?

File: webapp/src/components/CalendarPage.tsx [hunk 3]
```diff
@@ -1,7 +1,51 @@
+import { Container } from '@mui/material';
 import React from 'react';
+import useApiData from '../helpers/useApiData';
+
+import { ProgramWithActivities } from '../types/api';
+import Program from './Program';
 
 const CalendarPage = () => {
-  return <div>Calendar Page</div>;
+  const {
```
Comment (camianderson) [inline webapp/src/components/CalendarPage.tsx:9]: Great job calling the API.

File: webapp/src/components/CalendarPage.tsx (hunk 1 above)
Comment (wendyyng) [reply webapp/src/components/CalendarPage.tsx]: Just pushed a commit that tightens up the whitespace.

**Review by camianderson — APPROVED** [review]
Great job team!
The code looks great and its running correctly in my local machine.
I am approving it to merge.

File: webapp/src/components/Activity.tsx (hunk 2 above)
Comment (seidior) [reply webapp/src/components/Activity.tsx:3]: Good question! I don't think so. I'm not sure we're going to end up keeping this file anyway, but it seems the standard we have in other files is to have the Props structure in-file with the component.

**Pull request merged**